packages/contracts/contracts/votings/majority-voting/WhitelistVoting.sol

```diff
@@ -0,0 +1,175 @@
+/*
+ * SPDX-License-Identifier:    MIT
+ */
+
+pragma solidity 0.8.10;
+
+import "./MajorityVoting.sol";
+
+contract WhitelistVoting is MajorityVoting {
+    bytes32 public constant MODIFY_WHITELIST = keccak256("MODIFY_WHITELIST");
+
+    uint64 private whitelistedLength;
+
+    mapping(address => bool) public whitelisted;
+
+    error VoteCreationForbidden(address sender);
+
+    event AddUsers(address[] users);
+    event RemoveUsers(address[] users);
+
+    /// @dev describes the version and contract for GSN compatibility.
+    function versionRecipient() external view virtual override returns (string memory) {
+        return "0.0.1+opengsn.recipient.WhitelistVoting";
+    }
+
+    /// @dev Used for UUPS upgradability pattern
+    /// @param _dao The DAO contract of the current DAO
+    function initialize(
+        IDAO _dao,
+        address _gsnForwarder,
+        uint64 _participationRequiredPct,
+        uint64 _supportRequiredPct,
+        uint64 _minDuration,
+        address[] calldata _whitelisted
+    ) public initializer {
+        __MajorityVotingBase_init(
+            _dao,
+            _gsnForwarder,
+            _participationRequiredPct,
+            _supportRequiredPct,
+            _minDuration
+        );
+
+        // add whitelisted users
+        _addWhitelistedUsers(_whitelisted);
+    }
+
+    /// @notice add new users to the whitelist.
+    /// @param _users addresses of users to add
+    function addWhitelistedUsers(address[] calldata _users) external auth(MODIFY_WHITELIST) {
+        _addWhitelistedUsers(_users);
+    }
+
+    /// @dev Internal function to add new users to the whitelist.
+    /// @param _users addresses of users to add
+    function _addWhitelistedUsers(address[] calldata _users) internal {
+        for (uint256 i = 0; i < _users.length; i++) {
+            whitelisted[_users[i]] = true;
+        }
+
+        whitelistedLength += uint64(_users.length);
+
+        emit AddUsers(_users);
+    }
+
+    /// @notice remove new users to the whitelist.
+    /// @param _users addresses of users to remove
+    function removeWhitelistedUsers(address[] calldata _users) external auth(MODIFY_WHITELIST) {
+        for (uint256 i = 0; i < _users.length; i++) {
```

Comment:
don't want to annoy but if we anyways touch this could we directly put it into the ``unchecked`` closure and store the users length on the stack to save some gas.

Reply:
Added closure